packages/block-library/src/button/edit.js

```diff
@@ -173,19 +165,18 @@ class ButtonEdit extends Component {
 					</PanelBody>
 				</InspectorControls>
 				{ isSelected && (
-					<URLPopover
-						focusOnMount={ false }
-						position={ popoverPosition }
-					>
-						<form
-							className="editor-format-toolbar__link-container-content"
-							onSubmit={ ( event ) => event.preventDefault() }>
+					<form
+						className="editor-format-toolbar__url-inline"
+						onSubmit={ ( event ) => event.preventDefault() }>
+						<BaseControl label={ __( 'Link' ) }>
```

Comment:
There's a bit of work to do to make sure the label works correctly for screenreaders. Usually you'd have something like this for plain html:
```
<label for="wp-block-button__link">Link</label>
<input id="wp-block-button__link"  type="url" />
```
Where the label's `for` attribute matches the input's `id`. With the react components used here it's slightly different. `BaseControl` has an `id` prop that should match the `id` of the URLInput. Also, to protect against any chance of multiple instances of the same id, a unique number is added to the id.

Unfortunately it also looks like `URLInput` doesn't handle an `id` prop, so that will have to be modified.

The steps will be:
1. Make sure URLInput can receive an `id` prop that can be passed onto its `input` element:
https://github.com/WordPress/gutenberg/blob/5d14c988e4239baafebb35600f2f56e8a6a18952/packages/block-editor/src/components/url-input/index.js#L226-L248

2. Add the `withInstanceId` higher order component to the button's edit component here:
https://github.com/WordPress/gutenberg/blob/master/packages/block-library/src/button/edit.js#L147

(you can see an example of how `withInstanceId` is used to generate a unique id on the CheckboxControl - https://github.com/WordPress/gutenberg/blob/master/packages/components/src/checkbox-control/index.js#L12)

3. Create an id `const` (something like `wp-block-button__link-${ instanceId }`) and pass it to both `BaseControl` and `URLInput` as an `id` prop.

Ping me if you have any questions on that, I know it's a bit of work to achieve something fairly simple.

Reply:
Addressed this [here](https://github.com/WordPress/gutenberg/pull/10128/commits/57856208394951b6191204fdf6d130f6dac155a1) and [here](https://github.com/WordPress/gutenberg/pull/10128/commits/2ad43021a0273662c0ff5ea993634e301e9d2426). Looking forward to your review. Thank you ❤️